c++/src/kj/async.h

```diff
@@ -362,6 +362,25 @@ class Promise: protected _::PromiseBase {
   // the error handler if you are sure that ignoring errors is fine, or if you know that you'll
   // eventually wait on the promise somewhere.
 
+  Promise<T> adoptEnvironment() KJ_WARN_UNUSED_RESULT;
+  // Create a new promise which drops the existing promise's set of Environments and adopts the set
+  // of Environments that are currently active. This function is required inside of a
+  // `kj::runInEnvironment(env, func)`'s `func()` callback when importing promises from outside of
+  // `func()`'s scope. For example:
+  //
+  //   double environment = 1.23;
+  //   kj::Promise<void> outerPromise = foo();
+  //   kj::runInEnvironment(environment, [&]() {
+  //     return outerPromise.adoptEnvironment()
+  //                        .then([]() {
+  //       kj::getEnvironment<double>();
+  //     });
+  //   });
+  //
+  // In the above code, if the `.adoptEnvironment()` call were missing, the call to `.then()` would
+  // throw an exception in debug mode. (In release mode, that check is omitted, and the call to
+  // `kj::getEnvironment<double>()` would throw instead.)
```

Comment:
To clarify, would it throw because it detects misuse, or would it throw because it inherited the wrong environment and `double` doesn't exist in the environment?